deploy/test/test_cases/TEST_ID_25_helm_default_retry_successful.sh

```diff
@@ -22,29 +22,28 @@ pushd ../../
   DEFAULT_RETRY_COUNT_LIMIT=5
 
   fill_helm_chart_test_image
+
   fill_helm_chart_no_override_defaults
+
   helm install -f "../helm/secrets-provider/ci/take-default-test-values.yaml" \
     -f "../helm/secrets-provider/ci/take-image-values.yaml" \
     secrets-provider ../helm/secrets-provider \
     --set-file environment.conjur.sslCertificate.value="test/test_cases/conjur.pem"
 popd
 
-$cli_with_timeout "get pods --namespace=$APP_NAMESPACE_NAME --selector app=test-helm --no-headers"
+$cli_with_timeout "get pods --namespace=$APP_NAMESPACE_NAME --selector app=test-helm --no-headers | wc -l | tr -d ' ' | grep '^1$'"
 pod_name=$($cli_with_timeout get pods --namespace=$APP_NAMESPACE_NAME --selector app=test-helm --no-headers | awk '{print $1}' )
 
-# This sleep allows all logs to arrive to avoid platform failures
-sleep 5
-
 # Find initial authentication error that should trigger the retry
+SPACER=1
 $cli_with_timeout "logs $pod_name | grep 'CSPFK010E Failed to authenticate'"
-
-# Start the timer for retry interval
-start=$SECONDS
+failure_time=$($cli_without_timeout logs $pod_name | grep 'CSPFK010E Failed to authenticate' | head -1 | awk '{ print $3 }' | awk -F: '{ print ($1 * 3600) + ($2 * 60) + $3 }')
 
 echo "Expecting Secrets Provider retry configurations to take defaults RETRY_INTERVAL_SEC $DEFAULT_RETRY_INTERVAL_SEC and RETRY_COUNT_LIMIT $DEFAULT_RETRY_COUNT_LIMIT"
 $cli_with_timeout "logs $pod_name | grep 'CSPFK010I Updating Kubernetes Secrets: 1 retries out of $DEFAULT_RETRY_COUNT_LIMIT'"
+retry_time=$($cli_without_timeout logs $pod_name | grep "CSPFK010I Updating Kubernetes Secrets: 1 retries out of $DEFAULT_RETRY_COUNT_LIMIT" | head -1 | awk '{ print $3 }' | awk -F: '{ print ($1 * 3600) + ($2 * 60) + $3 }')
```

Comment:
can't we run it in one line which contain cli_with_timeout and will update the variable? like we have for example in test_id_18:
`pod_name=$($cli_with_timeout get pods --namespace=$APP_NAMESPACE_NAME --selector app=test-env --no-headers | awk '{print $1}' | head -1)`

Reply:
it does not account for the timeout. I rather leave it as is